src/dotnet/commands/dotnet-run/LaunchSettings/LaunchSettingsManager.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.Cli.Utils;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.Tools.Run.LaunchSettings
+{
+    internal class LaunchSettingsManager
+    {
+        private const string ProfilesKey = "profiles";
+        private const string CommandNameKey = "commandName";
+        private const string DefaultProfileCommandName = "Project";
+        private static IReadOnlyDictionary<string, ILaunchSettingsProvider> _providers;
+
+        static LaunchSettingsManager()
+        {
+            _providers = new Dictionary<string, ILaunchSettingsProvider>
+            {
+                { ProjectLaunchSettingsProvider.CommandNameValue, new ProjectLaunchSettingsProvider() }
+            };
+        }
+
+        public static LaunchSettingsApplyResult TryApplyLaunchSettings(string launchSettingsJsonContents, ref ICommand command, string profileName = null)
```

Comment:
Should we do all this JObject manipulation here? Wouldn't it be easier to have a proper object on which to deserialize the json and then work on top of that object model?